### ISSUE TITLE
Consider prefetching API data on page load

### ISSUE THREAD

**Issue by dechov** [Automattic/woocommerce-payments#728]
At the cost of skipping the loading placeholder view, we could **speed up loading time on initial page load** by pre<s>fetch</s>loading the data that will be needed by the view, in order to avoid round-trips on what I'd expect to be the most latency-prone leg of the stack, browser to site.

Currently, descriptors of what data is needed to populate the view are trapped deep inside the JS bundle, which must first be received and evaluated by the browser, making it inaccessible to the PHP server rendering piece.

However, we can write an adapter to derive required resources from the route and query params – see [prototype](https://github.com/Automattic/woocommerce-payments/commit/1c527064a43c4db1a866c8998579d6a72d1c1d2c) implementing one possible prefetching scheme for a few routes.

In many cases, the query params can be mapped directed to what the REST API handler expects (as is already done in client-side hooks [[code](https://github.com/Automattic/woocommerce-payments/blob/de82b68dee1e2233b8951065906c589ee8e0d3d4/client/data/transactions/hooks.js#L33)] and resolvers [[code](https://github.com/Automattic/woocommerce-payments/blob/de82b68dee1e2233b8951065906c589ee8e0d3d4/client/data/transactions/resolvers.js#L47)]). It looks like this will be good enough for all current routes except possibly when advanced filters are present, which I haven't attempted to prefetch due to the complexity of the mapping.

(It would be nice to have framework-level support for this but the routing logic [[code](https://github.com/woocommerce/woocommerce-admin/blob/d2fab15cb2b6e8cce8fd258e0d07580241334e9f/src/PageController.php#L116-L122)] doesn't currently even support patterns so we cannot do `/dispute/:dispute_id` even though the bundled client-side routing library fully supports it.)

Other considerations:
- One advantage of fetching the data from JS rather than PHP is that multiple requests are made in parallel – is there a way to do this from PHP without adding too much complexity?
- Since some of the same mapping logic would be implemented in the plugin PHP as in the browser code, we'd want to be sure they stayed in sync rather than drifting apart by accident – bidirectional references in code comments, perhaps
- Are there any good alternative ways of determining which requests will be needed by the view, or having the same specification be ingested in both contexts?
- Should the page query params be made to match the API query params more closely, to simplify the mapping? AFAIK the only hard limitation here is that other query params (`page`, in particular) are reserved, but we could make explicit exceptions for mapping those (just that one?)
- Would be worth doing some timing comparisons, primary consideration being the _time to load all data_, with _time to first UI render_ as another concern. (Some casual testing with a stopwatch _seems_ to indicate a quicker loading experience even with the deposit details screen preloading 3 resources in series)

**Comment by jessy-p** [comment]
Closing for now, based on comment paJDYF-5Cj-p2#comment-14218
